ddtrace/contrib/internal/openai/utils.py

```diff
@@ -429,25 +490,50 @@ def _compute_completion_tokens(completions_or_messages, model_name):
     return estimated, num_completion_tokens
 
 
-def _tag_tool_calls(integration, span, tool_calls, choice_idx):
+def _tag_tool_calls(integration, span, tool_calls, choice_idx=None):
     # type: (...) -> None
     """
-    Tagging logic if function_call or tool_calls are provided in the chat response.
+    Tagging logic if function_call, tool_calls in the chat response, or tools are provided in the responses.
     Notes:
         - since function calls are deprecated and will be replaced with tool calls, apply the same tagging logic/schema.
         - streamed responses are processed and collected as dictionaries rather than objects,
           so we need to handle both ways of accessing values.
     """
-    for idy, tool_call in enumerate(tool_calls):
-        if hasattr(tool_call, "function"):
-            # tool_call is further nested in a "function" object
-            tool_call = tool_call.function
-        function_arguments = _get_attr(tool_call, "arguments", "")
-        function_name = _get_attr(tool_call, "name", "")
-        span.set_tag_str(
-            "openai.response.choices.%d.message.tool_calls.%d.arguments" % (choice_idx, idy),
-            integration.trunc(str(function_arguments)),
-        )
-        span.set_tag_str(
-            "openai.response.choices.%d.message.tool_calls.%d.name" % (choice_idx, idy), str(function_name)
-        )
+    # Handle response tools
+    if hasattr(tool_calls, "tools"):
+        response_tools = []
+        for tool in tool_calls.tools:
+            if not (hasattr(tool, "type") or hasattr(tool, "name")):
+                continue
+
+            tool_dict = {}
+            if hasattr(tool, "type"):
+                tool_dict["type"] = getattr(tool, "type")
+            if hasattr(tool, "name"):
+                tool_dict["name"] = getattr(tool, "name")
+
+            if tool_dict:
+                response_tools.append(tool_dict)
+
+        # Only set the tag if we have valid tools to report
+        if response_tools:
+            span.set_tag("openai.response.tools", response_tools)
+    # Handle tool_calls (function calls and tool calls)
+    if choice_idx is not None and tool_calls and tool_calls != []:
```

Comment:
```suggestion
    return
    # Handle tool_calls (function calls and tool calls)
```
If we only care about this block for responses, we can just return early instead of making another conditional check to handle completions/chats tool calls.